src/Core/Geographic/Coordinates.js

```diff
@@ -1,441 +1,270 @@
- * coordinates.y(); // Geocentric system
- * // returns : 849926.376770819
- *
- * @return     {number} - The latitude of the position.
- */
+    set altitude(value) {
+        this.z = value;
+    }
 
-Coordinates.prototype.y = function y() {
-    _assertIsMetric(this.crs);
-    return this._values[1];
-};
+    get geodesicNormal() {
+        if (this._normalNeedsUpdate) {
+            this._normalNeedsUpdate = false;
 
-/**
- * Returns the altitude in geocentric coordinates. Coordinates must be in geocentric system (can be converted by using {@linkcode as()} ).
- * @example
- *
- * const position = { x: 20885167, y: 849862, z: 23385912 };
- * const coordinates = new Coordinates('EPSG:4978', position.x, position.y, position.z);  // Geocentric system
- * coordinates.z();  // Geocentric system
- * // returns : 23385912
- *
- * // or
- *
- * const position = { longitude: 2.33, latitude: 48.24, altitude: 24999549 };
- * const coords = new Coordinates('EPSG:4326', position.longitude, position.latitude, position.altitude); // Geographic system
- * const coordinates = coords.as('EPSG:4978'); // Geocentric system
- * coordinates.z(); // Geocentric system
- * // returns : 23382883.536591515
- *
- * @return     {number} - The altitude of the position.
- */
+            if (CRS.is4326(this.crs)) {
+                ellipsoid.geodeticSurfaceNormalCartographic(this, this._normal);
+            } else if (this.crs == 'EPSG:4978') {
+                ellipsoid.geodeticSurfaceNormal(this, this._normal);
+            } else {
+                this._normal.set(0, 0, 1);
+            }
+        }
 
-Coordinates.prototype.z = function z() {
-    _assertIsMetric(this.crs);
-    return this._values[2];
-};
+        return this._normal;
+    }
 
-/**
- * Returns a position in cartesian coordinates. Coordinates must be in geocentric system (can be converted by using {@linkcode as()} ).
- * @example
- *
- * const position = { x: 20885167, y: 849862, z: 23385912 };
- * const coordinates = new Coordinates('EPSG:4978', position.x, position.y, position.z);  // Geocentric system
- * coordinates.xyz();  // Geocentric system
- * // returns : Vector3
- * // x: 20885167
- * // y: 849862
- * // z: 23385912
- *
- * // or
- *
- * const position = { longitude: 2.33, latitude: 48.24, altitude: 24999549 };
- * const coords = new Coordinates('EPSG:4326', position.longitude, position.latitude, position.altitude); // Geographic system
- * const coordinates = coords.as('EPSG:4978'); // Geocentric system
- * coordinates.xyz(); // Geocentric system
- * // returns : Vector3
- * // x: 20885167
- * // y: 849862
- * // z: 23385912
- *
- * @return     {Position} - position
- */
+    /**
+     * Return this Coordinates values into a `THREE.Vector3`.
+     *
+     * @param {THREE.Vector3} [target] - The target to put the values in. If not
+     * specified, a new vector will be created.
+     *
+     * @return {THREE.Vector3}
+     */
+    xyz(target = new THREE.Vector3()) {
```

Comment:
May be rename by `toVector3` and to do mandatory the target, because `Coordinates` have `x`, `y` and `z` yet.
It should be seen in the code, when `Coordinates.xyz()` is used, if `Vector3` is needed.

Reply:
Done